invenio_rdm_records/services/services.py

```diff
@@ -295,3 +295,15 @@ def delete_secret_link(
         self._index_related_records(record, parent)
 
         return True
+
+    def reseve_doi(
```

Comment:
doubt: Are DOIs specific to RDM (I think so). Then it is the correct service to implement, them.